psp-examples/multisig/src/multisigParams.ts

```diff
@@ -0,0 +1,270 @@
+    this.priorMultiSigSlot = priorMultiSigSlot;
+    this.priorMultiSigHash = priorMultiSigHash;
+    this.priorMultiSigSeed = priorMultiSigSeed;
+  }
+
+  static async createNewMultiSig({
+    poseidon,
+    signers,
+    threshold,
+    randomSeed = nacl.randomBytes(32),
+  }: {
+    poseidon: any;
+    signers: Account[];
+    threshold: number;
+    randomSeed?: Uint8Array;
+  }) {
+    if (signers.length > MAX_SIGNERS) {
+      throw new Error(`Too many signers ${signers.length} > 6`);
+    }
+    const nrSigners = signers.length;
+    if (threshold > signers.length) {
+      throw new Error(
+        `Not enough signers ${signers.length} for threshold ${threshold}`,
+      );
+    }
+    if (poseidon === undefined) {
+      throw new Error("Poseidon instance not defined");
+    }
+    const dummyAccount = new Account({
+      poseidon,
+      seed: new Uint8Array(32).fill(3).toString(),
+    });
+    dummyAccount.poseidonEddsaKeypair = {
+      publicKey: [new Uint8Array(32).fill(0), new Uint8Array(32).fill(0)],
+      privateKey: new Uint8Array(32).fill(0),
+    };
+
+    let publicKeyX: Uint8Array[] = [];
+    let publicKeyY: Uint8Array[] = [];
+
+    for (let signer of signers) {
+      const pubkey = await signer.getEddsaPublicKey();
+      publicKeyX.push(pubkey[0]);
+      publicKeyY.push(pubkey[1]);
+    }
+
+    while (MAX_SIGNERS > signers.length) {
+      signers.push(dummyAccount);
+      // const pubkey = new Uint8Array(32).fill(0)
+      publicKeyX.push(new Uint8Array(32).fill(0));
+      publicKeyY.push(new Uint8Array(32).fill(0));
+    }
+
+    const signersEncryptionPublicKeys = signers.map(
+      (signer) => signer.encryptionKeypair.publicKey,
+    );
+
+    return new MultisigParams({
+      poseidon,
+      threshold,
+      nrSigners,
+      publicKeyX,
+      publicKeyY,
+      signersEncryptionPublicKeys,
+      seed: randomSeed,
+      priorMultiSigSlot: new BN(0),
+      priorMultiSigHash: new Uint8Array(32).fill(0),
+      priorMultiSigSeed: new Uint8Array(32).fill(0),
+    });
+  }
+
+  async toBytes(): Promise<Buffer> {
+    let coder = new BorshAccountsCoder(IDL);
+    return await coder.encode("createMultiSig", this);
+  }
+
+  static fromBytes(poseidon, bytes: Buffer): MultisigParams {
+    let coder = new BorshAccountsCoder(IDL);
+    let decoded = coder.decode("createMultiSig", bytes);
+    return new MultisigParams({ poseidon, ...decoded });
+  }
+
+  print() {
+    console.log("----------------- MultiSig Parameters -----------------");
+    console.log("threshold: ", this.threshold.toString());
+    console.log("Number of Signers: ", this.nrSigners.toString());
+    console.log(`Shielded pubkey: ${this.appDataHash}`);
+    console.log("Shared encryption public key: <encryption-key>");
+    console.log("Shared encryption private key: <encryption-key>");
+
+    for (var i = 0; i < this.publicKeyX.length; i++) {
+      if (i < this.nrSigners.toNumber()) {
```

Comment:
If we want to limit the loop to `this.nrSigners.toNumber()`, why don't we do this instead?

```suggestion
    for (var i = 0; i < Math.min(this.nrSigners.toNumber(), this.publicKeyX.length); i++) {
```